ui/src/app/intro/views/MaasIntro/MaasIntro.tsx

```diff
@@ -1,79 +1,123 @@
+import { actions as repoActions } from "app/store/packagerepository";
+import repoSelectors from "app/store/packagerepository/selectors";
 
-export const IntroSchema = Yup.object()
+export const MaasIntroSchema = Yup.object()
   .shape({
+    httpProxy: Yup.string().url("Must be a valid URL."),
+    mainArchiveUrl: Yup.string()
+      .url("Must be a valid URL.")
+      .required("Ubuntu archive is required."),
     name: Yup.string().required("MAAS name is required"),
+    portsArchiveUrl: Yup.string()
+      .url("Must be a valid URL.")
+      .required("Ubuntu extra architectures is required."),
+    upstreamDns: Yup.string(),
   })
   .defined();
 
 const MaasIntro = (): JSX.Element => {
   const dispatch = useDispatch();
   const authLoading = useSelector(authSelectors.loading);
-  const authUser = useSelector(authSelectors.get);
-  const errors = useSelector(configSelectors.errors);
+  const httpProxy = useSelector(configSelectors.httpProxy);
   const maasName = useSelector(configSelectors.maasName);
+  const upstreamDns = useSelector(configSelectors.upstreamDns);
+  const configErrors = useSelector(configSelectors.errors);
   const configLoading = useSelector(configSelectors.loading);
-  const saved = useSelector(configSelectors.saved);
-  const saving = useSelector(configSelectors.saving);
+  const configSaved = useSelector(configSelectors.saved);
+  const configSaving = useSelector(configSelectors.saving);
+  const reposErrors = useSelector(repoSelectors.errors);
+  const reposLoading = useSelector(repoSelectors.loading);
+  const reposSaving = useSelector(repoSelectors.saving);
+  const mainArchive = useSelector(repoSelectors.mainArchive);
+  const portsArchive = useSelector(repoSelectors.portsArchive);
 
   useWindowTitle("Welcome");
 
-  if (authLoading || configLoading) {
-    return <Spinner />;
-  }
+  useEffect(() => {
+    dispatch(repoActions.fetch());
+  }, [dispatch]);
+
+  const errors = { ...configErrors, ...reposErrors };
+  const loading = authLoading || configLoading || reposLoading;
+  const saving = configSaving || reposSaving;
+  const saved = configSaved;
 
   return (
     <Section>
-      <FormikForm<MaasIntroValues>
-        buttonsBordered={false}
-        cleanup={configActions.cleanup}
-        errors={errors}
-        initialValues={{
-          name: maasName || "",
-        }}
-        onSaveAnalytics={{
-          action: "Saved",
-          category: "Intro",
-          label: "Intro form",
-        }}
-        onSuccess={() => {
-          dispatch(configActions.fetch());
-        }}
-        onSubmit={(values) => {
-          dispatch(configActions.cleanup());
-          dispatch(
-            configActions.update({
-              completed_intro: true,
-              maas_name: values.name,
-            })
-          );
-        }}
-        savedRedirect={
-          authUser?.completed_intro
-            ? machineURLs.machines.index
-            : baseURLs.intro.user
-        }
```

Comment:
I've removed the redirect for now since there'll be an extra step for the images page

Reply:
Should it be changed to `introURLs.images` or will you do that in your other branch?

Reply:
I'll handle it in my other branch